pkg/utils/memory_resources.go

```
@@ -68,8 +68,8 @@ func GetNumaMemoryResources() (NumaMemoryResources, error) {
 
 		// Get hugepages
 		hugepageBytes, err := getHugepagesBytes(filepath.Join(sysBusNodeBasepath, numaNode, "hugepages"))
-		if err != nil {
-			return nil, err
+		if err != nil && os.IsNotExist(err) {
+			continue
```

We still want to return an error is other error cases

---

this may be the best chocies?😅
```
               // Get hugepages
		huagePagePath := filepath.Join(sysBusNodeBasepath, numaNode, "hugepages")
		_, err = os.Stat(huagePagePath)
		if err != nil && os.IsNotExist(err) {
			continue
		}
		hugepageBytes, err := getHugepagesBytes(huagePagePath)
```

---

How about using `else`?

---

Thank for your suggestion,@marquiz,is this ok?

---

Looks legit to me 👍